src/layouts/sidebar-sidecar/index.tsx

```diff
@@ -17,21 +18,90 @@ const SidebarSidecarLayout = ({
   sidebarProps,
   sidebarSlot,
   sidecarSlot,
+  sidebarPropsLevels,
 }: SidebarSidecarLayoutProps) => {
+  const { isDesktop } = useDeviceSize()
+  const numSidebarLevels = sidebarPropsLevels ? sidebarPropsLevels.length : 0
+  const initialSidebarLevel = useMemo(() => {
+    if (numSidebarLevels > 0) {
+      return numSidebarLevels - 1
+    } else {
+      return 0
+    }
+  }, [numSidebarLevels])
+
+  // If `sidebarPropsLevels` is given, set current level to last level.
+  const [currentSidebarLevel, setCurrentSidebarLevel] =
+    useState<number>(initialSidebarLevel)
+
+  /**
+   * @TODO I think this is temporary until all the pages have
+   * `sidebarPropsLevels`. Otherwise, when we switch from something like
+   * /waypoint to /waypoint/docs, the sidebar level that shows is the first one
+   * because this layout component isn't getting re-rendered in a way that
+   * re-initializes `currentSidebarLevel`.
+   */
+  useEffect(() => {
+    setCurrentSidebarLevel(initialSidebarLevel)
+  }, [initialSidebarLevel])
+
+  /**
+   * Initializes sidebarProps based on the current device size, number of
+   * sidebar props levels given, and current sidebar level.
+   */
+  const finalSidebarProps: SidebarProps = useMemo(() => {
+    // Make backwards compatible with `sidebarProps`
+    if (!sidebarPropsLevels) {
+      return sidebarProps
+    }
+
+    // Copy provided props so we don't modify the original
+    const propsCopy = {
+      ...sidebarPropsLevels[currentSidebarLevel],
+    }
+
+    /**
+     * If desktop, we only care about `backToLinkProps`. Otherwise, tell each
+     * level's Button what to do `onClick`.
+     */
+    if (isDesktop) {
+      delete propsCopy.levelButtonProps
+    } else {
+      delete propsCopy.backToLinkProps
+
+      if (numSidebarLevels > 1) {
+        propsCopy.levelButtonProps.onClick = () => {
+          setCurrentSidebarLevel((prevLevel: number) => {
+            if (prevLevel === 0) {
+              return prevLevel + 1
+            } else {
+              return prevLevel - 1
+            }
+          })
```

Comment:
Functional observation: Feels kinda funky that you can't navigate back to the docs menu once you click back 🤔

Reply:
I think this is good feedback! I feel the same. It _might_ feel less weird in a few PRs when `Sidebar` is actually collapsing and the main header navigation has a "close" icon button for dismissing the `Sidebar` when it's expanded. If we still feel like the experience is weird, I think we can raise it in the VQA process. Does that sound good?

Reply:
Non-blocking suggestion:

Personally I think this might be worth raising with design more immediately. It looks like in the `Mobile menu flow chart` that the expectation is that we'd be able to get back and forth between panels... I might be missing something, but it seems like there's no way to get back to the lowest level panel 💭 

Even with the "close" button, it seems like we'd have to close and then re-open the menu to get "back" to the lowest-level panel... and it seems possible that doing such a "panel reset" might not even happen naturally as a result of opening and closing the menu; it seems like we'd need an additional `useEffect` to reset `currentSidebarLevel` when the open-close state changes.

This feels like it might compromise the intent of the complex panelled approach, which I think is to allow exploration of available sidebar links without page navigation or extra clicks. It seems less like a VQA issue to me, and more like something we might need to re-evaluate from a user experience & design standpoint.

Maybe I'm being a bit dramatic here though... possibly the solution is as straightforward as adding a `Docs →` link at the top of the `{Product} Home` panel? Though I guess this link would need to be `Learn →` on Learn views 🤔

Reply:
You're not being dramatic! 😊  I'll go ahead and raise it via Slack now, and make sure we're all aligned on the intended functionality. Thank you for adding your +1 to this!

Reply:
We got this one figured out via Slack, so resolving! ✨